### ISSUE TITLE
Problems with aliases

### ISSUE THREAD

**Issue by parosio** [sscarduzio/elasticsearch-readonlyrest-plugin#424]
Hello, 

I'm testing 6.6.1 elastic with 1.17.0 ROR, and it seems that something got broken from 6.5 / 1.16.17

I've got several aliases which are used to group indices, like `all-subs-data`:
```
  curl "http://localhost:9202/_cat/aliases/all-subs-data?pretty" -H "X-Forwarded-User:PAROSIO"
  all-subs-data ib-blue-coat-sub-algeria                - - -
  all-subs-data ib-blue-coat-sub-angola                 - - -
  all-subs-data ib-blue-coat-sub-china                  - - -
  all-subs-data ib-blue-coat-sub-congo                  - - -
  all-subs-data ib-blue-coat-sub-eastafrica             - - -
  all-subs-data ib-blue-coat-sub-myanmar                - - -
  all-subs-data ib-blue-coat-sub-norge                  - - -
  all-subs-data ib-blue-coat-sub-northafrica            - - -
  all-subs-data ib-blue-coat-sub-ungheria               - - -
  all-subs-data ib-blue-coat-sub-vietnam                - - -
  all-subs-data sec-incidents-algeria                   - - -
  all-subs-data sec-incidents-angola                    - - -
  all-subs-data sec-incidents-china                     - - -
  all-subs-data sec-incidents-congo                     - - -
  all-subs-data sec-incidents-eastafrica                - - -
  all-subs-data sec-incidents-myanmar                   - - -
  all-subs-data sec-incidents-norge                     - - -
  all-subs-data sec-incidents-northafrica               - - -
  all-subs-data sec-incidents-tunisia                   - - -
  all-subs-data sec-incidents-vietnam                   - - -
  all-subs-data vuln-ass-all-algeria                    - - -
  all-subs-data vuln-ass-all-angola                     - - -
  all-subs-data vuln-ass-all-australia                  - - -
  all-subs-data vuln-ass-all-china                      - - -
  all-subs-data vuln-ass-all-congo                      - - -
  all-subs-data vuln-ass-all-eastafrica                 - - -
  all-subs-data vuln-ass-all-myanmar                    - - -
  all-subs-data vuln-ass-all-norge                      - - -
  all-subs-data vuln-ass-all-northafrica                - - -
  all-subs-data vuln-ass-all-vietnam                    - - -
```

when I try to access only a subset of the indices through the same alias using a less privileged account then mine I get a 401 error:

The account has the following acl in readonlyrest.yml:
```
    - name: myanmar_vietnam
      kibana_access: ro
      proxy_auth: [ "VIET_MYAN" ]
      indices: ["*myanmar*","*vietnam*", ".kibana*", ".kibana-devnull"]
```

The query failing:

```
curl "http://localhost:9202/all-subs-data/_search?pretty" -H "X-Forwarded-User:VIET_MYAN" -H 'Content-Type: application/json' -d'
> {
>   "size": 0,
>
>     "aggs": {
>       "2": {
>         "terms": {
>           "field": "_company",
>           "size": 5
>         }
>       }
>     }
> }'
{
  "error" : {
    "root_cause" : [
      {
        "reason" : "Forbidden by ELASTIC"
      }
    ],
    "reason" : "Forbidden by ELASTIC"
  },
  "status" : 401
}
```

``[2019-04-08T17:47:37,027][INFO ][t.b.r.a.ACL              ] [mCMhoMa] FORBIDDEN by default req={ ID:2078241775-696891403#285950, TYP:SearchRequest, CGR:N/A, USR:VIET_MYAN, BRS:true, KDX:null, ACT:indices:data/read/search, OA:127.0.0.1, DA:127.0.0.1, IDX:all-subs-data, MET:POST, PTH:/all-subs-data/_search?pretty, CNT:<OMITTED, LENGTH=140>, HDR:{Accept=*/*, User-Agent=curl/7.37.0, Host=localhost:9202, Content-Length=140, X-Forwarded-User=VIET_MYAN, Content-Type=application/json}, HIS:[::LOGSTASH::->[auth_key->false]], [::KIBANA-SRV::->[auth_key->false]], [::PHP::->[auth_key->false]], [::NAGIOS::->[auth_key->false]], [HQ Admins->[proxy_auth->false]], [HQ ReadOnly->[proxy_auth->false]], [Norge->[proxy_auth->false]], [eastafrica->[proxy_auth->false]], [myanmar_vietnam->[kibana_access->true, indices->false, proxy_auth->true]] }
``

When I go directly to indices ROR doesn't block the request:

```
curl "http://localhost:9202/*/_search?pretty" -H "X-Forwarded-User:VIET_MYAN" -H 'Content-Type: application/json' -d'
> {
>   "size": 0,
>
>     "aggs": {
>       "2": {
>         "terms": {
>           "field": "_company",
>           "size": 5
>         }
>       }
>     }
> }
> '
{
  "took" : 1,
  "timed_out" : false,
  "_shards" : {
    "total" : 8,
    "successful" : 8,
    "skipped" : 0,
    "failed" : 0
  },
  "hits" : {
    "total" : 8160,
    "max_score" : 0.0,
    "hits" : [ ]
  },
  "aggregations" : {
    "2" : {
      "doc_count_error_upper_bound" : 0,
      "sum_other_doc_count" : 0,
      "buckets" : [
        {
          "key" : "Myanmar BV",
          "doc_count" : 5214
        },
        {
          "key" : "Vietnam BV",
          "doc_count" : 2946
        }
      ]
    }
  }
}
```

Obviously, I need to access data using aliases, and not directly using index names.

This issue seems to be linked to [https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/issues/291](https://github.com/sscarduzio/elasticsearch-readonlyrest-plugin/issues/291)

In the logs of 5.4 the ROR log read: 
``[2019-04-08T18:02:01,992][INFO ][t.b.r.a.ACL              ] ALLOWED by { name: 'myanmar_vietnam', policy: ALLOW} req={ ID:961211522-347393503#3350, TYP:SearchRequest, CGR:N/A, USR:VIET_MYAN, BRS:true, ACT:indices:data/read/search, OA:127.0.0.1, DA:127.0.0.1, IDX:vuln-ass-all-myanmar,vuln-ass-all-vietnam,sec-incidents-myanmar,ib-blue-coat-sub-vietnam,vuln-ass-score-myanmar,ib-blue-coat-sub-myanmar,vuln-ass-score-vietnam,sec-incidents-vietnam, MET:POST, PTH:/all-subs-data/_search?pretty, CNT:
``
so, apparently, the unwrap of alias to indices was already done in this version.

**Comment by TRISAF** [comment]
Hi @sscarduzio any update ?
I found that also thereadonlyrest plugin for ElasticSearch 6.7.0 has the same bug.

**Comment by parosio** [comment]
Hi Mateusz, @coutoPL 
glad to see you managed to address and fix the aliases problems. 
I'm not familiar enough with Scala and Java to really understand your changes to the code.
The tests you added correctly represent the situation we've met, and I appreciate it's considered a bug.

Hi Simone, @sscarduzio: is it possible to have a build with this fix for Elastic 6.7.0?
